### ISSUE TITLE
In POSIX sh, umask flags besides -S are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1459]
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
umask -p
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
"In POSIX sh, umask -p is undefined"

The only allowed flag under POSIX sh for the `umask` builtin is `-S` ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/umask.html)). Although `man dash` doesn't show any flags for `umask`, `umask -S` does work under the latest dash (0.5.10.2). On the contrary, bash allows a `-p` flag too.
